packages/venia-concept/src/components/MiniCart/miniCart.js

```diff
@@ -134,73 +141,75 @@ class MiniCart extends Component {
-
-        const itemName = this.state.focusItem
-            ? this.state.focusItem.name
-            : null;
-        const itemPrice = this.state.focusItem
-            ? this.state.focusItem.price
-            : null;
-
-        return (
-            <div className={classes.content}>
-                <div className={classes.focusItem}>
-                    {itemName}
-                    <div className={classes.price}>${itemPrice}</div>
-                </div>
-                <div className={classes.options}>Choose a Size:</div>
-            </div>
-        );
-    }
-
-    get productConfirm() {
-        const { classes } = this.props;
-
-        return (
-            <div className={classes.save}>
-                <Button onClick={this.hideEditPanel}>Cancel</Button>
-                <Button priority="high">Update Cart</Button>
-            </div>
+        const { props, state, hideEditPanel } = this;
+        const { updateItemInCart } = props;
+        const { focusItem } = state;
+
+        if (focusItem === null) return;
+        const hasOptions = focusItem.options.length !== 0;
+
+        return hasOptions ? (
+            // `Name` is being used here because GraphQL does not allow
+            // filtering products by id, and sku is unreliable without
+            // a reference to the base product. Additionally, `url-key`
+            // cannot be used because we don't have page context in cart.
+            <Query
+                query={getProductDetailByName}
+                variables={{ name: focusItem.name, onServer: false }}
+            >
+                {({ loading, error, data }) => {
+                    if (error) return <div>Data Fetch Error</div>;
+                    if (loading) return <div>Fetching Data</div>;
+
+                    const itemWithOptions = data.products.items[0];
+
+                    return (
+                        <Fragment>
+                            <CartOptions
+                                cartItem={focusItem}
+                                configItem={itemWithOptions}
+                                hideEditPanel={hideEditPanel}
+                                updateCart={updateItemInCart}
+                            />
+                        </Fragment>
+                    );
+                }}
+            </Query>
+        ) : (
+            <Fragment>
+                <CartOptions
+                    cartItem={focusItem}
+                    configItem={{}}
+                    hideEditPanel={hideEditPanel}
+                    updateCart={updateItemInCart}
+                />
+            </Fragment>
         );
     }
 
     showEditPanel = item => {
         this.setState({
-            isEditPanelOpen: true,
             focusItem: item
         });
+        this.props.openEditPanel();
     };
 
     hideEditPanel = () => {
-        this.setState({
-            isEditPanelOpen: false
-        });
+        this.props.hideEditPanel();
```

Comment:
Please add `openEditPanel` and `hideEditPanel` to this Component's `propTypes` and make them `required` since we're calling them directly.